docs/detailed_examples.rst

```diff
@@ -0,0 +1,144 @@
+6.	Choose a type of contour: IFormContour, ISormContour, DirectSamplingContour or HighestDensityContour.
+
+50 year environmental contour with the Hs-Tz DNVGL model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here, we use a sea state dataset with the variables Hs and Tz, fit the join distribution recommended in
+DNVGL-RP-C203 [1]_ to it and compute an IFORM contour. This example reproduces the results published in
+Haselsteiner et al. (2019) [2]_. Such a work flow is for example typical in ship design. The presented example can be
+downloaded from the examples_ section of the repository.
+
+**Imports**
+::
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    from virocon import (
+    read_ec_benchmark_dataset,
+    GlobalHierarchicalModel,
+    WeibullDistribution,
+    LogNormalDistribution,
+    DependenceFunction,
+    WidthOfIntervalSlicer,
+    IFORMContour,
+    plot_marginal_quantiles,
+    plot_dependence_functions,
+    plot_2D_contour,
+    )
+
+**Environmental data**
+
+This dataset has been used in a benchmarking exercise, see https://github.com/ec-benchmark-organizers/ec-benchmark .
+The dataset was derived from NDBC buoy 44007, see https://www.ndbc.noaa.gov/station_page.php?station=44007 .
+::
+    data = read_ec_benchmark_dataset("datasets/ec-benchmark_dataset_A.txt")
+
+**Dependence structure**
+
+Define the structure of the joint model that we will use to describe the environmental data. To define a joint model, we
+define the univariate parametric distributions and the dependence structure. The dependence structure is defined using
+parametric functions.
+
+A 3-parameter power function, which will be used as a dependence function.
+::
+    def _power3(x, a, b, c):
+        return a + b * x ** c
+A 3-parameter exponential function, which will be used a dependence function.
+::
+    def _exp3(x, a, b, c):
+        return a + b * np.exp(c * x)
+
+
+**Parametric distributions**
+First, set lower and upper interval boundaries for the three parameter values.
+::
+    bounds = [(0, None), (0, None), (None, None)]
+
+    power3 = DependenceFunction(_power3, bounds)
+    exp3 = DependenceFunction(_exp3, bounds)
+
+    dist_description_0 = {
+        "distribution": WeibullDistribution(),
+        "intervals": WidthOfIntervalSlicer(width=0.5),
+    }
+    dist_description_1 = {
+        "distribution": LogNormalDistribution(),
+        "conditional_on": 0,
+        "parameters": {"mu": power3, "sigma": exp3}
+
+**Multivariate Model**
+::
+    model = GlobalHierarchicalModel([dist_description_0, dist_description_1])
+
+Semantics define a dictionary that describes the model. This description can be used while plotting the contour.
+::
+    semantics = {
+    "names": ["Significant wave height", "Zero-crossing period"],
+    "symbols": ["H_s", "T_z"],
+    "units": ["m", "s"],
+    }
+
+Fit the model to the data (estimate the model's parameter values).
```

Comment:
The following  sentences and snippets are rather short. I think it might help the readers flow if the snippets where a little larger and if there where a little more text.